src/tests/ascent/t_ascent_expressions.cpp

```diff
@@ -44,628 +44,939 @@
+  EXPECT_EQ(res["type"].as_string(), "double");
+}
 
-    multi_dom.child(0)["state/cycle"] = 300;
+//-----------------------------------------------------------------------------
+TEST(ascent_expressions, if_expressions)
+{
+  Node n;
+  ascent::about(n);
+
+  //
+  // Create an example mesh.
+  //
+  Node data, verify_info;
+  conduit::blueprint::mesh::examples::braid("hexs",
+                                            EXAMPLE_MESH_SIDE_DIM,
+                                            EXAMPLE_MESH_SIDE_DIM,
+                                            EXAMPLE_MESH_SIDE_DIM,
+                                            data);
+  // ascent normally adds this but we are doing an end around
+  data["state/domain_id"] = 0;
+  Node multi_dom;
+  blueprint::mesh::to_multi_domain(data, multi_dom);
+
+  runtime::expressions::register_builtin();
+  runtime::expressions::ExpressionEval eval(&multi_dom);
+
+  conduit::Node res;
+  std::string expr;
+
+  expr = "if 5 == 5 then 1 else 2";
+  res = eval.evaluate(expr);
+  EXPECT_EQ(res["value"].to_int32(), 1);
+  EXPECT_EQ(res["type"].as_string(), "int");
+
+  bool threw = false;
+  try
+  {
+    expr = "if max(3,7) > min(2,6) then 1 else vector(1,1,1)";
+    res = eval.evaluate(expr);
+  }
+  catch(...)
+  {
+    threw = true;
+  }
+  EXPECT_EQ(threw, true);
+}
 
-    res = eval.evaluate("3", "val");
-    res = eval.evaluate("vector(3,4,0)", "vec");
+//-----------------------------------------------------------------------------
+TEST(ascent_binning, binning_basic_meshes)
+{
+  // the vtkm runtime is currently our only rendering runtime
+  Node n;
+  ascent::about(n);
+  // only run this test if ascent was built with vtkm support
+  if(n["runtimes/ascent/vtkm/status"].as_string() == "disabled")
+  {
+    ASCENT_INFO("Ascent support disabled, skipping test");
+    return;
+  }
+
+  //
+  // Create an example mesh.
+  //
+  Node data, verify_info;
+
+  conduit::blueprint::mesh::examples::basic("hexs", 3, 3, 3, data);
+
+  // ascent normally adds this but we are doing an end around
+  data["state/cycle"] = 100;
+  data["state/domain_id"] = 0;
+  Node multi_dom;
+  blueprint::mesh::to_multi_domain(data, multi_dom);
+
+  runtime::expressions::register_builtin();
+  runtime::expressions::ExpressionEval eval(&multi_dom);
+
+  conduit::Node res;
+  std::string expr;
+
+  expr = "binning('field', 'sum', [axis('x', [0, 2.5, 5, 7.5, 10])])";
+  res = eval.evaluate(expr);
+  EXPECT_EQ(res["attrs/value/value"].to_json(), "[0.0, 0.0, 16.0, 0.0]");
+
+  expr = "binning('field', 'max', [axis('z', [-5, 0, 5])])";
+  res = eval.evaluate(expr);
+  EXPECT_EQ(res["attrs/value/value"].to_json(), "[3.0, 0.0]");
+
+  expr = "binning('field', 'max', [axis('z', [-5, 0, 5], clamp=True)])";
+  res = eval.evaluate(expr);
+  EXPECT_EQ(res["attrs/value/value"].to_json(), "[3.0, 7.0]");
+
+  expr =
+      "binning('field', 'max', [axis('x', num_bins=4), axis('y', num_bins=4)], "
```

Comment:
I like how this turned out! 
Q: Is there a way to do something like:

`axis('x", num_bins=4, min=0, max = 1)`

Reply:
Yes `axis('x', num_bins=4, min_val=0, max_val=1)`

Reply:
One thing I should say is that the bins won't be inclusive on max_val.